packages/core/test/after.test.ts

```diff
@@ -43,6 +43,30 @@ describe('delayed transitions', () => {
     expect(actorRef.getSnapshot().value).toBe('yellow');
   });
 
+  it('should never try to clear an undefined timeout ID', () => {
```

Comment:
why did this one fail before the change? how did that `undefined` creep in there? 🤔

Reply:
There's 2 places where the ID can be deleted:

```ts
 const scheduler: Scheduler = {
    schedule: (
      // ...
    ) => {
      // ...

      const timeout = clock.setTimeout(() => {
        delete timerMap[scheduledEventId]; // here
        delete system._snapshot._scheduledEvents[scheduledEventId];

        system._relay(source, target, event);
      }, delay);

      timerMap[scheduledEventId] = timeout;
    },
    cancel: (source, id: string) => {
      // ...

      delete timerMap[scheduledEventId]; // and here
      delete system._snapshot._scheduledEvents[scheduledEventId];

      if (timeout !== undefined) {
        clock.clearTimeout(timeout);
      }
    },
```
If the timeout happens before `cancel(…)` happens, the ID is already deleted, hence `undefined`

Reply:
ah, ok - cause the cancellation itself is not canceled in a sense when the timeout completes. This makes sense 👍